2-mandatory/6-writing-tests-advanced.js

```diff
@@ -73,19 +86,53 @@ function formatCourseworkResult(trainee) {
     subjects: ["JavaScript", "React", "CSS"]
   }
 */
-
+test("Trainee name Ali, score E", () => {
+  let trainee = {
+    name: `Ali`,
+    score: 49,
+    age: 33,
+    subjects: ["JavaScript", "React", "CSS"],
+  };
+  expect(formatCourseworkResult(trainee)).toEqual(
+    `Ali's coursework was marked as grade E.`
+  );
+});
 /*
   Write a test that checks the output of formatCourseworkResult when passed the following trainee:
   {
     score: 90,
     age: 29
   }
 */
-
+test("Trainee name undefined , score A", () => {
+  let trainee = {
+    age: 29,
+    score: 90,
+  };
+  expect(formatCourseworkResult(trainee)).toEqual(`Error: No trainee name!`);
+});
 /*
   Write a test that checks the output of formatCourseworkResult when passed the following trainee:
   {
     name: "Aman",
     subjects: ["HTML", "CSS", "Databases"]
   }
 */
```

Comment:
Well done on all of these. Just remember to think about the best way to name your tests 😄 
Imagine you make a change to your code - and this test starts to fail. In the test output, you will see `Trainee name Aman` - but this won't give you any useful information about which part of the code might be broken. Can you think of a better name for this?